src/AppBundle/Controller/DonationController.php

```diff
@@ -21,8 +24,50 @@ class DonationController extends Controller
      */
     public function indexAction(Request $request)
     {
+        $amount = $request->query->get('montant');
+        $isMonthlyPayment = $request->get('abonnement');
+
+        if ($amount) {
```

Comment:
Can you make the opposite to avoid nesting conditions:

```php
if (!$amount = $request->query->get('montant')) {
    return $this->render('...');
}

if ($request->[query|request|attributes]->get('abonnement', false)) {
    return $this->redirectToRoute('donation_frequency', ['montant' => $amount]);
}

// ...
```